tests/radiation_transport/test_radiation_profile.py

```diff
@@ -296,3 +305,49 @@ def test_radiative_loss_function_values(self):
         lvals = np.array([imp_data_l_ref[1], imp_data_l_ref[3]])
         l1 = radiative_loss_function_values(tvals, t_ref, l_ref)
         np.testing.assert_allclose(l1, lvals, rtol=2e-1)
+
+    def test_pfr_filter(self):
+        x_point_z = self.source.sol_rad.points["x_point"]["z_low"]
+        pfr_x_down, pfr_z_down = pfr_filter(self.source.sol_rad.separatrix, x_point_z)
+        assert pfr_x_down.shape == (59,)
+        assert pfr_z_down.shape == (59,)
+
+        assert np.all(pfr_z_down < x_point_z - 0.01)
+
+    def test_make_wall_detectors(self):
+        max_wall_len = 10.0e-2
+        X_WIDTH = 0.01
+        wall_detectors = make_wall_detectors(
+            self.fw_shape.x, self.fw_shape.z, max_wall_len, X_WIDTH
+        )
+        assert all(detector[2] <= max_wall_len for detector in wall_detectors)
+        assert all(detector[1] == X_WIDTH for detector in wall_detectors)
+        assert len(wall_detectors) == 532
+
+    def test_FirstWallRadiationSolver(self):
+        # Coversion required for CHERAB
+        f_sol = linear_interpolator(
+            self.source.sol_rad.x_tot,
+            self.source.sol_rad.z_tot,
+            raw_uc(self.source.sol_rad.rad_tot, "MW", "W"),
+        )
+
+        # SOL radiation grid
+        x_sol = np.linspace(min(self.fw_shape.x), max(self.fw_shape.x), 4)
+        z_sol = np.linspace(min(self.fw_shape.z), max(self.fw_shape.z), 4)
+
+        rad_sol_grid = interpolated_field_values(x_sol, z_sol, f_sol)
+        func = grid_interpolator(x_sol, z_sol, rad_sol_grid)
+        solver = FirstWallRadiationSolver(
+            source_func=func, firstwall_shape=self.fw_shape
+        )
+        assert solver.fw_shape == self.fw_shape
+
+        wall_loads = solver.solve(50, 1, 10, plot=False)
+
+        # check return types
+        assert isinstance(wall_loads, DetectedRadiation)
+        assert isinstance(wall_loads.total_power, float)
+
+        # check if the arrays are of same length
+        assert len(wall_loads.detector_numbers) == len(wall_loads.detected_power)
```

Comment:
the asserts here are no thorough enough for a test. At the moment it is just essentially checking it runs which is not enough for a good test.

Can you add some more information that we can assert on

Reply:
The problem is, the solver do not always give exact same results (e.g. wall_loads.total_power gives different values in different runs). I have now added an assertion to check if the wall_loads.total_power gives a value of a certain order.